### PR TITLE
Promotion slot slots

### PR DESCRIPTION
**Pull request by albertgcsula**
* Update promotion-details-fields to include `slots` schema field.

### CODE REVIEW

File: src/plugins/curator/components/promotion-details-fields/Slots.jsx [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+  render() {
+    const { fields, readOnly, borderClass } = this.props;
+
+    return (
+      <ul className="list-group list-group-flush" ref={(el) => { this.sortableListRef = el; }}>
+        <ListGroupItem className="p-0 pb-2">
+          {!readOnly && (
+            <Button color="primary" onClick={() => fields.push({ _id: UuidIdentifier.generate() })}>
+              <Icon imgSrc="plus" alt="+" className="mr-1" />Add a Slot
```
Comment (gdbrown) [inline src/plugins/curator/components/promotion-details-fields/Slots.jsx:82]: @albertgcsula slots don't have an _id, the

File: src/plugins/curator/components/promotion-details-fields/Slots.jsx [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+          {!readOnly && (
+            <Button color="primary" onClick={() => this.handleAdd()}>
+              <Icon imgSrc="plus" alt="+" className="mr-1" />Add a Slot
+            </Button>
+          )}
+        </ListGroupItem>
+
+        {fields.getAll().map((slot, index) => (
+          <ListGroupItem
+            style={{"z-index": String(99 - index)}}
```
Comment (arunkarnati) [inline src/plugins/curator/components/promotion-details-fields/Slots.jsx:105]: @albertgcsula - is there a real need for the `z-index` rule here? I'm getting this warning in the devTools:
`Unsupported style property z-index. Did you mean zIndex?`

File: src/plugins/curator/components/promotion-details-fields/Slots.jsx [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+            fields.push();
+          }
+        });
+      },
+    });
+  }
+
+  handleAdd() {
+    const { fields } = this.props;
+    fields.push({ name: "" });
```
Comment (arunkarnati) [inline src/plugins/curator/components/promotion-details-fields/Slots.jsx:58]: @albertgcsula - nit: use single quotes for strings

File: src/plugins/curator/components/promotion-details-fields/Slots.jsx (hunk 2 above)
Comment (albertgcsula) [reply src/plugins/curator/components/promotion-details-fields/Slots.jsx]: @arunkarnati fixed

File: src/plugins/curator/components/promotion-details-fields/Slots.jsx (hunk 3 above)
Comment (albertgcsula) [reply src/plugins/curator/components/promotion-details-fields/Slots.jsx]: fixed

**Pull request merged**